packages/commerce-sdk-react/src/hooks/ShopperBaskets/query.ts

```diff
@@ -1,172 +1,234 @@
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperbaskets.shopperbaskets-1.html#getbasket} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-function useBasket(
-    arg: Omit<UseBasketArg, 'rawResponse'> & {rawResponse?: false},
-    options?: UseQueryOptions<DataType<Client['getBasket']> | Response, Error>
-): UseQueryResult<DataType<Client['getBasket']>, Error>
-function useBasket(
-    arg: Omit<UseBasketArg, 'rawResponse'> & {rawResponse: true},
-    options?: UseQueryOptions<DataType<Client['getBasket']> | Response, Error>
-): UseQueryResult<Response, Error>
-function useBasket(
-    arg: UseBasketArg,
-    options?: UseQueryOptions<DataType<Client['getBasket']> | Response, Error>
-): UseQueryResult<DataType<Client['getBasket']> | Response, Error> {
-    const {headers, rawResponse, ...parameters} = arg
-    const defaultOptions = {
-        enabled: !!parameters.basketId
-    }
-    return useQuery(
-        ['/baskets', parameters.basketId, arg],
-        (_, {shopperBaskets}) => shopperBaskets.getBasket({parameters, headers}, rawResponse),
-        {...defaultOptions, ...options}
-    )
-}
+export const useBasket = (
+    apiOptions: Argument<Client['getBasket']>,
+    queryOptions: Omit<UseQueryOptions<DataType<Client['getBasket']>>, 'queryFn'> = {}
+): UseQueryResult<DataType<Client['getBasket']>> => {
+    const {shopperBaskets: client} = useCommerceApi()
+    const method = (arg: Argument<Client['getBasket']>) => client.getBasket(arg)
+    const requiredParameters = ['organizationId', 'basketId', 'siteId'] as const
+    // Parameters can be set in `apiOptions` or `client.clientConfig`; they are merged in the helper
+    // hook, so we use a callback here that receives that merged object.
+    const getQueryKey = <T extends Record<string, unknown>>(parameters: T) =>
+        [
+            'https://',
+            parameters.shortCode,
+            '.api.commercecloud.salesforce.com/checkout/shopper-baskets/',
+            parameters.version,
+            '/organizations/',
+            parameters.organizationId,
+            '/baskets/',
+            parameters.basketId,
+            '?',
+            'siteId',
+            parameters.siteId,
+            'locale',
+            parameters.locale,
+            // Full parameters last for easy lookup
+            parameters
+        ] as const
 
-type UsePaymentMethodsForBasketParameters = NonNullable<
-    Argument<Client['getPaymentMethodsForBasket']>
->['parameters']
-type UsePaymentMethodsForBasketHeaders = NonNullable<
-    Argument<Client['getPaymentMethodsForBasket']>
->['headers']
-type UsePaymentMethodsForBasketArg = {
-    headers?: UsePaymentMethodsForBasketHeaders
-    rawResponse?: boolean
-} & UsePaymentMethodsForBasketParameters
+    return useQuery(apiOptions, queryOptions, {
```

Comment:
I am not a fan of the new signature of the `useQuery` here, I find it hard to use practically. Previously, we write the `useQuery` to mimic the signature of React query `useQuery<queryKey, QueryKeyFn, options>`. Now the signatures change to useQuery<apiOptions, queryOptions, configOptions>, which is entirely unfamiliar. 🤔 . I am wondering if it brings more benefits with the new signatures

Reply:
The new signature of our `useQuery` helper is designed to mimic the signature of our endpoint query hooks (`useBasket(apiOptions, queryOptions)`), rather than the signature of react query's `useQuery`. Doing it this way makes it easier to keep the login in our helper `useQuery`, rather than in the endpoint hooks.